spec/requests/registrations_spec.rb

```diff
@@ -1,28 +1,46 @@
 # frozen_string_literal: true
 
+# rubocop:disable Metrics/BlockLength
 RSpec.describe 'Registrations', type: :request do
-
-  shared_examples "sucessful password update" do
-    let(:update_params) { { password: new_password, password_confirmation: new_password } }
+  shared_examples 'successfully updates user and redirects' do
     before do
       sign_in user
       put user_registration_path, params: update_params
+      user.reload
     end
 
     it { expect(response).to redirect_to(edit_user_registration_path) }
+    it { expect(user.comment_notify).to be(false) }
   end
 
-  describe "update password" do
-    let(:new_password) { "foobar_w!th_@" }
+  describe '#update' do
+    let(:update_params) { { user: { comment_notify: 'false' } } }
 
-    context "when the user is logged in with Google OAuth" do
+    context 'when the user is logged in with Google OAuth' do
       let(:user) { create(:user_oauth, provider: 'google_oauth2') }
-      it_behaves_like "sucessful password update"
+      it_behaves_like 'successfully updates user and redirects'
     end
 
-    context "when the user is logged in with Facebook" do
+    context 'when the user is logged in with Facebook' do
       let(:user) { create(:user_oauth, provider: 'facebook') }
-      it_behaves_like "sucessful password update"
+      it_behaves_like 'successfully updates user and redirects'
+    end
+
+    context 'when the user is logged in' do
+      let(:user) { create(:user, provider: nil) }
+      let(:update_params) do
+        { user: { name: user.name, email: user.email, password: new_password, password_confirmation: new_password,
+                  current_password: user.password, comment_notify: 'false' } }
+      end
+      let(:new_password) { 'foobar_w!th_@' }
+      it_behaves_like 'successfully updates user and redirects'
+
+      it 'successfully changes password' do
+        sign_in user
+        put user_registration_path, params: update_params
+        user.reload
```

Comment:
I think we're missing the assertion here `expect` command, but I'd probably ditch the test altogether base on my thoughts below.

```suggestion
      it 'successfully changes password' do
        sign_in user
        expect { put user_registration_path, params: update_params }.to change { user.password }
```

---

While trying this myself, I feel like there's a problem here and that the current `update_params` password isn't going to change. I feel like we're trying to test Devise implementation here which we should expect and trust that the gem tests the password change implementation itself. 

You'd have to look at [Devise gem](https://github.com/heartcombo/devise) itself here to see exactly what `update_params` to use. My question would be why would this be necessary? This is something that I'd expect Devise to test itself. From what I remember one time, I forgot if it was the password or password confirmation that was hashed and passed on the way (you'll have to reread Devise implementation again), but that'd be overkill for this implementation test.

Reply:
I agree with you. We are going too way beyond the issue's scope. The only test here should be if the action is redirecting properly. I am going to delete all the unnecessary testing. 
Regarding your last point, I have look into the actual request's payload (deploying locally the app) and the payload was exactly the way I set it up. Either way, as discussed above this is out of scope.